scripts/Modules/agc_lib.py

```diff
@@ -0,0 +1,297 @@
+    file = open(filename, 'w')
+    if file is None :
+        gui.add_osd_message("Error : could not create the data file")
+    else :
+        file.write(str(metadata))
+        file.close()
+        gui.add_osd_message(f"{filename} successfully opened")
+
+def get_metadata(readid):
+    return FrameData(get_metadata_addr(readid))
+
+def get_rkg_metadata():
+    return FrameData(get_rkg_metadata_addr())
+
+def rkg_metadata_to_file(filename):
+    rkg_metadata = get_rkg_metadata()
+    file = open(filename, 'w')
+    if file is None :
+        gui.add_osd_message("Error : could not create the data file")
+    else :
+        file.write("r"+str(rkg_metadata))
+        file.close()
+        gui.add_osd_message(f"{filename} successfully opened")   
+
+def frame_to_file(filename, readid):
+    frame = FrameData(get_addr(readid))
+    file = open(filename, 'a')
+    if file is None :
+        gui.add_osd_message("Error : could not create the data file")
+    else :
+        file.write(str(frame))
+        file.close()
+
+def get_framedata(readid):
+    return FrameData(get_addr(readid))
+    
+def timerdata_to_file(filename, rid):
+    timerdata = TimerData(readid = rid)
+    file = open(filename, 'a')
+    if file is None :
+        gui.add_osd_message("Error : could not create the data file")
+    else :
+        file.write(str(timerdata))
+        file.close()
+
+def get_timerdata(rid):
+    return TimerData(readid = rid)
+        
+def file_to_framedatalist(filename):
+    datalist = []
+    file = open(filename, 'r')
+    if file is None :
+        gui.add_osd_message("Error : could not load the data file")
+    else:
+        timerdata = None
+        metadata = None
+        rkg_metadata = None
+        listlines = file.readlines()
+        if listlines[0][0] == 'r':           
+            rkg_metadata = FrameData(string = listlines[0][1:])
+            timerdata = TimerData.from_sliced_rkg(rkg_metadata)
+        else:
+            metadata = FrameData(string = listlines[0])
+            if listlines[-1][0]=='t':
+                timerdata = TimerData(string = listlines.pop()[1:])
+        for i in range(1, len(listlines)):
+            datalist.append(FrameData(string = listlines[i]))
+        file.close()
+        gui.add_osd_message(f"Data successfully loaded from {filename}")
+        return metadata, datalist, timerdata, rkg_metadata
+
+
+def framedatalist_to_file(filename, datalist, rid):
+    metadata = get_metadata(rid)
+    timerdata = get_timerdata(rid)
+    file = open(filename, 'w')
+    if file is None :
+        gui.add_osd_message("Error : could not create the data file")
+    else:
+        file.write(str(metadata))
+        for frame in range(max(datalist.keys())+1):
+            if frame in datalist.keys():
+                file.write(str(datalist[frame]))
+            else:
+                file.write(str(FrameData(get_addr(rid), usedefault=True)))
+        file.write(str(timerdata))
+        file.close()
```

Comment:
I think sticking to just `read_id` or `rid` would be best here, unless this is necessary for it to work.

Reply:
> I think sticking to just `read_id` or `rid` would be best here, unless this is necessary for it to work.

```suggestion
def timerdata_to_file(filename, read_id):
    timerdata = TimerData(read_id)
    file = open(filename, 'a')
    if file is None :
        gui.add_osd_message("Error : could not create the data file")
    else :
        file.write(str(timerdata))
        file.close()

def get_timerdata(read_id):
    return TimerData(read_id)
        
def file_to_framedatalist(filename):
    datalist = []
    file = open(filename, 'r')
    if file is None :
        gui.add_osd_message("Error : could not load the data file")
    else:
        timerdata = None
        metadata = None
        rkg_metadata = None
        listlines = file.readlines()
        if listlines[0][0] == 'r':           
            rkg_metadata = FrameData(string = listlines[0][1:])
            timerdata = TimerData.from_sliced_rkg(rkg_metadata)
        else:
            metadata = FrameData(string = listlines[0])
            if listlines[-1][0]=='t':
                timerdata = TimerData(string = listlines.pop()[1:])
        for i in range(1, len(listlines)):
            datalist.append(FrameData(string = listlines[i]))
        file.close()
        gui.add_osd_message(f"Data successfully loaded from {filename}")
        return metadata, datalist, timerdata, rkg_metadata


def framedatalist_to_file(filename, datalist, read_id):
    metadata = get_metadata(read_id)
    timerdata = get_timerdata(read_id)
    file = open(filename, 'w')
    if file is None :
        gui.add_osd_message("Error : could not create the data file")
    else:
        file.write(str(metadata))
        for frame in range(max(datalist.keys())+1):
            if frame in datalist.keys():
                file.write(str(datalist[frame]))
            else:
                file.write(str(FrameData(get_addr(rid), usedefault=True)))
        file.write(str(timerdata))
        file.close()
```